base/util.jl

```diff
@@ -429,6 +429,7 @@ function julia_cmd(julia=joinpath(Sys.BINDIR::String, julia_exename()))
 end
 
 function julia_exename()
+    @static Base.DARWIN_FRAMEWORK && return "julia"
```

Comment:
This doesn't seem necessary to me; why not just leave it to call `jl_is_debugbuild`, then return `"julia"`?

Reply:
I refactored the code and now BUNDLE_DEBUG_LIBS=1 causes julia-debug (effectively linked to libjulia-debug) to be included in the framework.